src/dndbeyond/content-scripts/character.js

```diff
@@ -1153,27 +1179,48 @@ async function rollAction(paneClass, force_to_hit_only = false, force_damages_on
-            });
+
+        const applyGWFIfRequired = (damageDie) => {
+            // Polearm master bonus attack using the other end of the polearm is considered a melee attack.
+            // check for both versions
+            if (action_name.includes("Polearm Master") && character.hasGreatWeaponFighting(2014)) {
+                damageDie = damageDie.replace(/[0-9]*d[0-9]+/g, "$&ro<=2");
+            } else if (action_name.includes("Polearm Master") && character.hasGreatWeaponFighting(2024)) {
+                damageDie = damageDie.replace(/([0-9]*)d([0-9]+)([^\s+-]*)(.*)/g, (match, amount, faces, roll_mods, mods) => {
+                    return new Array(parseInt(amount) || 1).fill(`1d${faces}${roll_mods}min3`).join(" + ") + mods;
+                });
+            }
+
+            // check for both versions
+            if (action_name.includes("Pole Strike") && (character.hasFeat("Polearm Master 2024", false)) && 
+                character.hasGreatWeaponFighting(2024)) {
+                damageDie = damageDie.replace(/([0-9]*)d([0-9]+)([^\s+-]*)(.*)/g, (match, amount, faces, roll_mods, mods) => {
+                    return new Array(parseInt(amount) || 1).fill(`1d${faces}${roll_mods}min3`).join(" + ") + mods;
+                });
+            } else if (action_name.includes("Pole Strike") && (character.hasFeat("Polearm Master 2024", false)) && 
+                character.hasGreatWeaponFighting(2014)) {
+                damageDie = damageDie.replace(/[0-9]*d[0-9]+/g, "$&ro<=2");
+            }
+
+            return damageDie;
         }
 
-        // check for both versions
-        if (action_name.includes("Pole Strike") && (character.hasFeat("Polearm Master 2024", false)) && 
-            character.hasGreatWeaponFighting(2024)) {
-            damages[0] = damages[0].replace(/([0-9]*)d([0-9]+)([^\s+-]*)(.*)/g, (match, amount, faces, roll_mods, mods) => {
-                return new Array(parseInt(amount) || 1).fill(`1d${faces}${roll_mods}min3`).join(" + ") + mods;
-            });
-        } else if (action_name.includes("Pole Strike") && (character.hasFeat("Polearm Master 2024", false)) && 
-            character.hasGreatWeaponFighting(2014)) {
-            damages[0] = damages[0].replace(/[0-9]*d[0-9]+/g, "$&ro<=2");
+        // 2024 brutal strike add extra die and apply GWF
+        if (character.getSetting("brutal-strike")) {
+            let strikeDieCount = character.hasClassFeature("Brutal Strike") ? 1 : 0;
+            if (character.hasClassFeature("Improved Brutal Strike")) strikeDieCount += 1;
+
+            let brutal_strike_dmg = applyGWFIfRequired(`${strikeDieCount}d10`);
+
+            damages.push(brutal_strike_dmg);
+            damage_types.push("Brutal Strike");
+
+            settings_to_change["brutal-strike"] = false;
         }
 
+        // apply for normal die
+        damages[0] = applyGWFIfRequired(damages[0]);
```

Comment:
I see you thought of factoring the code out into a function after all. I also prefer the name you used vs the one I proposed in my previous comment 😁